elyra/pipeline/processor_kfp.py

```diff
@@ -52,6 +57,29 @@ class KfpPipelineProcessor(RuntimePipelineProcess):
     def type(self):
         return self._type
 
+    @property
+    def component_registry(self) -> str:
+        return self._component_registry_location
+
+    @property
+    def component_parser(self) -> ComponentParser:
+        return self._component_parser
+
+    def __init__(self, root_dir, **kwargs):
+        super().__init__(root_dir, **kwargs)
+        # then sys.prefix, where installed files will reside (factory data)
+        self._component_registry_location =  \
+            os.path.join(jupyter_core.paths.ENV_JUPYTER_PATH[0],
+                         'components',
+                         f"{self._type}_component_catalog.json")
+
+        if not os.path.exists(self._component_registry_location):
+            raise FileNotFoundError(f'Invalid component_id catalog path {self._component_registry_location}'
+                                    f' for {self._type} processor')
+
+        self._component_parser = KfpComponentParser()
+        self._component_registry = CachedComponentRegistry(self.component_registry, self.component_parser)
```

Comment:
What is the plan for allowing users to not use registry caching?  Seems like this needs to be configurable somehow.